### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by alexisluque**
* Add dockerfile and scripts to run it on Linux and Windows.
* Update README.md

### CODE REVIEW

File: 01-Login/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM gradle:4.2-jdk8
```
Comment (lbalmaceda) [inline 01-Login/Dockerfile]: use jdk7 as we still support it.

File: 01-Login/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+WORKDIR /tmp
+ADD . /tmp
+
+RUN gradle build
+
+CMD java -jar build/libs/auth0-spring-mvc.war
```
Comment (lbalmaceda) [inline 01-Login/Dockerfile]: We [are telling users](https://github.com/auth0-samples/auth0-spring-mvc-sample/tree/master/01-Login#running-the-sample) to run the sample using `bootRun` as that's the task that spring boot adds to the project. What's the benefit on doing it this other way?

File: 01-Login/README.md [hunk 3]
```diff
@@ -42,6 +42,13 @@ Open a terminal, go to the project root directory and run the following command:
 
 The server will be accessible on https://localhost:8080/portal/home. After logging in you should see the `token` in the header.
 
+### Running the sample with docker
+
+In order to run the example with docker you need to have `docker` installed.
```
Comment (lbalmaceda) [inline 01-Login/README.md]: I'd add a link to the [installation guide](https://docs.docker.com/engine/installation/)

File: 01-Login/README.md [hunk 4]
```diff
@@ -42,6 +42,13 @@ Open a terminal, go to the project root directory and run the following command:
...
+
+You also need to set the client values as explained [previously](#java-application).
+
+Execute in command line `sh exec.sh` to run the Docker in Linux, or `.\exec.ps1` to run the Docker in Windows.
```
Comment (lbalmaceda) [inline 01-Login/README.md]: Linux or MacOS

File: 01-Login/exec.ps1 [hunk 5]
```diff
@@ -0,0 +1,2 @@
+docker build -t auth0-spring-mvc-webapp .
```
Comment (lbalmaceda) [inline 01-Login/exec.ps1]: Would be nice to tag them under a "samples" category with a "latest" version tag. I was using the next one:

`auth0-samples/{REPO}-{PATH}:latest` -> `auth0-samples/spring-mvc-00login:latest`

**Review by lbalmaceda — CHANGES_REQUESTED** [review]

File: 01-Login/exec.ps1 [hunk 6]
```diff
@@ -1,2 +1,2 @@
-docker build -t auth0-spring-mvc-webapp .
-docker run -p 8080:8080 -it auth0-spring-mvc-webapp
+docker build -t auth0-samples/spring-mvc-00login:latest .
```
Comment (lbalmaceda) [inline 01-Login/exec.ps1]: In some repos you've used just the repo name for the tags. We should be consistent with that.

File: 01-Login/exec.sh [hunk 7]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
-docker build -t auth0-spring-mvc-webapp .
-docker run -p 8080:8080 -it auth0-spring-mvc-webapp
+docker build -t auth0-samples/spring-mvc-00login:latest .
```
Comment (lbalmaceda) [inline 01-Login/exec.sh]: See the other comment

**Review by lbalmaceda — CHANGES_REQUESTED** [review]

File: 01-Login/exec.ps1 [hunk 8]
```diff
@@ -0,0 +1,2 @@
...
+docker run -p 3000:3000 -v ~/.m2:/home/gradle/.gradle -it auth0-samples/spring-mvc-00login:latest
```
Comment (lbalmaceda) [inline 01-Login/exec.ps1]: this -v option is mounting the local directory `~/.m2` into the container folder `/home/gradle/.gradle`. Hosts should not need to install anything other than Docker.

File: 01-Login/exec.ps1 [hunk 9]
```diff
@@ -0,0 +1,2 @@
+docker build -t auth0-samples/spring-mvc-00login:latest .
```
Comment (lbalmaceda) [inline 01-Login/exec.ps1]: Keep the tag name consistency across repos

File: README.md [hunk 10]
```diff
@@ -1 +1,37 @@
...
+Please do not report security vulnerabilities on the public GitHub issue tracker.
+The [Responsible Disclosure Program](https://auth0.com/whitehat) details the procedure for disclosing security issues.
+
+## Author
+
+[Auth0](https://auth0.com)
+
+## License
+
+This project is licensed under the MIT license. See the [LICENSE](LICENSE) file for more info.
```
Comment (lbalmaceda) [inline README.md:37]: File is named `LICENCE`

**Review by lbalmaceda — CHANGES_REQUESTED** [review]

File: 01-Login/exec.ps1 (hunk 8 above)
Comment (alexisluque) [reply 01-Login/exec.ps1]: The gradle image is defining a volume in `/home/gradle/.gradle` directory, if we don't mount the volume dependencies installation won't be cached.

**Review by lbalmaceda — DISMISSED** [review]

File: 01-Login/exec.ps1 (hunk 8 above)
Comment (lbalmaceda) [reply 01-Login/exec.ps1]: There's no need to cached them. But if you think the opposite, adding a `RUN gradle clean build` command to the docker file would make them already available in the image on the run time, right?

File: 01-Login/exec.sh [hunk 11]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+docker build -t auth0-spring-mvc-01-login .
```
Comment (lbalmaceda) [inline 01-Login/exec.sh]: rename to `auth0-samples/auth0-spring-mvc-01-login`

File: 01-Login/exec.ps1 [hunk 12]
```diff
@@ -0,0 +1,2 @@
+docker build -t auth0-spring-mvc-01-login .
+docker run -p 3000:3000 -v ~/.m2:/home/gradle/.gradle -it auth0-spring-mvc-01-login
```
Comment (lbalmaceda) [inline 01-Login/exec.ps1]: `auth0-samples/auth0-spring-mvc-01-login`

**Review by lbalmaceda — APPROVED** [review]

**Review by aaguiarz — APPROVED** [review]

**Pull request merged**